src/unix/linux_like/linux_l4re_shared.rs

```diff
@@ -0,0 +1,2103 @@
+        pub updated: c_ulong,
+        pub ha: [c_uchar; crate::MAX_ADDR_LEN],
+    }
+
+    pub struct regmatch_t {
+        pub rm_so: crate::regoff_t,
+        pub rm_eo: crate::regoff_t,
+    }
+
+    pub struct option {
+        pub name: *const c_char,
+        pub has_arg: c_int,
+        pub flag: *mut c_int,
+        pub val: c_int,
+    }
+
+    pub struct rlimit64 {
+        pub rlim_cur: rlim64_t,
+        pub rlim_max: rlim64_t,
+    }
+
+    pub struct __c_anonymous_ifru_map {
+        pub mem_start: c_ulong,
+        pub mem_end: c_ulong,
+        pub base_addr: c_ushort,
+        pub irq: c_uchar,
+        pub dma: c_uchar,
+        pub port: c_uchar,
+    }
+}
+
+cfg_if! {
+    if #[cfg(not(target_arch = "sparc64"))] {
+        s! {
+            pub struct __c_anonymous_elf32_rela {
+                pub r_offset: crate::Elf32_Addr,
+                pub r_info: crate::Elf32_Word,
+                pub r_addend: crate::Elf32_Sword,
+            }
+
+            pub struct __c_anonymous_elf64_rela {
+                pub r_offset: crate::Elf64_Addr,
+                pub r_info: crate::Elf64_Xword,
+                pub r_addend: crate::Elf64_Sxword,
+            }
+        }
+    }
+}
+
+s_no_extra_traits! {
+    pub struct dirent {
+        pub d_ino: crate::ino_t,
+        pub d_off: crate::off_t,
+        pub d_reclen: c_ushort,
+        pub d_type: c_uchar,
+        pub d_name: [c_char; 256],
+    }
+
+    pub struct dirent64 {
+        pub d_ino: crate::ino64_t,
+        pub d_off: crate::off64_t,
+        pub d_reclen: c_ushort,
+        pub d_type: c_uchar,
+        pub d_name: [c_char; 256],
+    }
+
+    pub union __c_anonymous_ifr_ifru {
+        pub ifru_addr: crate::sockaddr,
+        pub ifru_dstaddr: crate::sockaddr,
+        pub ifru_broadaddr: crate::sockaddr,
+        pub ifru_netmask: crate::sockaddr,
+        pub ifru_hwaddr: crate::sockaddr,
+        pub ifru_flags: c_short,
+        pub ifru_ifindex: c_int,
+        pub ifru_metric: c_int,
+        pub ifru_mtu: c_int,
+        pub ifru_map: crate::__c_anonymous_ifru_map,
+        pub ifru_slave: [c_char; crate::IFNAMSIZ],
+        pub ifru_newname: [c_char; crate::IFNAMSIZ],
+        pub ifru_data: *mut c_char,
+    }
+
+    pub struct ifreq {
+        /// interface name, e.g. "en0"
+        pub ifr_name: [c_char; crate::IFNAMSIZ],
+        pub ifr_ifru: __c_anonymous_ifr_ifru,
+    }
+
+    pub union __c_anonymous_ifc_ifcu {
+        pub ifcu_buf: *mut c_char,
+        pub ifcu_req: *mut crate::ifreq,
+    }
+
+    /// Structure used in SIOCGIFCONF request.  Used to retrieve interface configuration for
+    /// machine (useful for programs which must know all networks accessible).
+    pub struct ifconf {
+        /// Size of buffer
+        pub ifc_len: c_int,
+        pub ifc_ifcu: __c_anonymous_ifc_ifcu,
+    }
+}
+
+#[cfg(not(target_env = "uclibc"))]
+const base: crate::nl_item = 0x20000;
+#[cfg(target_env = "uclibc")]
+const base: crate::nl_item = 768;
```

Comment:
Give this a name like DATE_BASE so its less conflict prone. Also `if cfg!(...)` is preferable over `#[cfg(...)]` where possible, so it gets checked on more targets.